solutions/platinum/cses-1742.mdx

```diff
@@ -0,0 +1,246 @@
+---
+id: cses-1742
+source: CSES
+title: Robot Path
+author: Mihnea Brebenel
+---
+
+## Explanation
+As you have already guessed this is a sweep line problem.
+But the main issue is how to find the first intersection when the robot stops ? To find the first intersection, we use binary search.
+
+The last thing to do is to find the length of the incomplete segment - the on that causes the intersection. There are several ways to do this.
+The easiest way is to binary search for it.
+
+## Implementation
+
+**Time Complexity: $\mathcal O(N \log N)$**
```

Comment:
I believe that this should be $\mathcal{O}(n\log^2 n)$ because of the binary search.

Reply:
Ok I changed it. With the segment tree operation it's $\log^2 N$